### ISSUE TITLE
Record<A> GetHashCode calculating wrong values

### ISSUE THREAD

**Issue by tanxeel90** [louthy/language-ext#560]
GetHashCode for record type is not considering the 'name' of the property while calculating hashcodes so if two properties within a record have same type, the hashcode is same for the record.

example:

```C#        
        public class TestRecord : Record<TestRecord>
        {
            public readonly Option<int> Opt1;
            public readonly Option<int> Opt2;

            public TestRecord(Option<int> opt1, Option<int> opt2)
            {
                Opt1 = opt1;
                Opt2 = opt2;
            }
        }
        static void Main(string[] args)
        {
            var testRecord1 = new TestRecord(Option<int>.None, 2);
            var testRecord2 = new TestRecord(2, Option<int>.None);

            var isHashCodeEqual = testRecord1.GetHashCode() == testRecord2.GetHashCode(); //TRUE: expected to be false

            testRecord1 = new TestRecord(Option<int>.None, Option<int>.None);
            testRecord2 = new TestRecord(1, 1);

            isHashCodeEqual = testRecord1.GetHashCode() == testRecord2.GetHashCode(); //TRUE: expected to be FALSE
        }
```

**Comment by bender2k14** [comment]
These hash codes are colliding because the members are being [XORed together](https://github.com/louthy/language-ext/blob/master/LanguageExt.Core/Utility/IL.cs#L447).

@louthy, can we change the implementation of `Record<>.GetHashCode()` to follow Josh Bloch's 
 recommendation in [Effective Java](https://www.amazon.com/Effective-Java-Joshua-Bloch/dp/0134685997) (3rd Edition, Item 11, Always override hashCode when you override equals)?  This would change the delegate in `Fold` to (effectively) `(state, field) => 31 * state + field`.

**Comment by louthy** [comment]
Fixed in v3.3.28 thanks @alaendle
